src/s2ir.c

```diff
@@ -1100,7 +1102,26 @@ class S2irVisitor : public Visitor
 
                 assert(cs->type);
 
-                Symbol *catchtype = toSymbol(cs->type->toBasetype());
+                /* The catch type can be a C++ class or a D class.
+                 * If a D class, insert a pointer to TypeInfo into the typesTable[].
+                 * If a C++ class, insert a pointer to __cpp_type_info_ptr into the typesTable[].
+                 */
+                Type *tcatch = cs->type->toBasetype();
+                ClassDeclaration *cd = tcatch->isClassHandle();
+                bool isCPPclass = cd->isCPPclass();
+                Symbol *catchtype;
+                if (isCPPclass)
+                {
+                    catchtype = toSymbolCpp(cd);
+                    if (i == 0)
+                    {
+                        // rewrite ebegin to use __cxa_begin_catch
+                        Symbol *s = getRtlsym(RTLSYM_CXA_BEGIN_CATCH);
+                        ebegin->EV.sp.Vsym = s;
```

Comment:
This is a more FYI than anything else, but you can assume that ``__cxa_begin_catch`` will never throw.  So if there's anything you can do to optimize for that, the better.